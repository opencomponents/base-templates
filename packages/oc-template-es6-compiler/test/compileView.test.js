jest.setTimeout(50000);

const path = require('path');
const fs = require('fs-extra');
const compileView = require('../lib/compileView.js');

test('valid component', done => {
  const componentPath = path.join(__dirname, '../../../mocks/es6-component');
  const publishPath = path.join(componentPath, '_packageCompileViewTest');
  const publishFileName = 'template.js';

  const options = {
    componentPackage: fs.readJsonSync(`${componentPath}/package.json`),
    componentPath,
    publishPath,
    publishFileName,
    production: true
  };

  compileView(options, (err, compiledViewInfo) => {
    const viewHashKey = compiledViewInfo.template.hashKey;
    compiledViewInfo.template.hashKey = 'dummyData';
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs
        .readFileSync(path.join(publishPath, publishFileName), 'UTF8')
        .replace(viewHashKey, 'dummyData')
    ).toMatchSnapshot();
    fs.removeSync(publishPath);
    done();
  });
});

test('invalid component', done => {
  const componentPath = path.join(__dirname, '../../../mocks/es6-component');
  const publishPath = path.join(componentPath, '_packageCompileViewTest2');
  const publishFileName = 'template.js';
  const componentPackage = fs.readJsonSync(`${componentPath}/package.json`);
  componentPackage.oc.files.template.src = 'broken.js';

  const options = {
    componentPackage,
    componentPath,
    publishPath,
    publishFileName,
    production: true
  };

  compileView(options, (err, compiledViewInfo) => {
    expect(err).toContain('SyntaxError: Unexpected token (2:2)');
    done();
  });
});
