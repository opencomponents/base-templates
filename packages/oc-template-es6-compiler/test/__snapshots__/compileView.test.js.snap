// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid component 1`] = `
Object {
  "template": Object {
    "hashKey": "dummyData",
    "src": "template.js",
    "type": "oc-template-es6",
  },
}
`;

exports[`valid component 2`] = `"var oc=oc||{};oc.components=oc.components||{};oc.components['dummyData']=function(model){ var template = (function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.d=function(a,c,d){b.o(a,c)||Object.defineProperty(a,c,{enumerable:!0,get:d})},b.r=function(a){'undefined'!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(a,Symbol.toStringTag,{value:'Module'}),Object.defineProperty(a,'__esModule',{value:!0})},b.t=function(a,c){if(1&c&&(a=b(a)),8&c)return a;if(4&c&&'object'==typeof a&&a&&a.__esModule)return a;var d=Object.create(null);if(b.r(d),Object.defineProperty(d,'default',{enumerable:!0,value:a}),2&c&&'string'!=typeof a)for(var e in a)b.d(d,e,function(b){return a[b]}.bind(null,e));return d},b.n=function(a){var c=a&&a.__esModule?function(){return a['default']}:function(){return a};return b.d(c,'a',c),c},b.o=function(a,b){return Object.prototype.hasOwnProperty.call(a,b)},b.p='',b(b.s=2)})([function(a){a.exports={awesome:'awesome__p4GK2pPl'}},,function(a,b,c){'use strict';c.r(b);var d=c(0),e=c.n(d);b['default']=function(a){var b=a.name;return'<div class='+e.a.awesome+'>Hello '+b.toUpperCase()+'!</div>'}}]);; return '' + template.default(model) + '<style>.awesome__p4GK2pPl{font-family:\\"Hiragino Sans\\",\\"ヒラギノ角ゴシック\\",\\"Hiragino Kaku Gothic ProN\\",\\"ヒラギノ角ゴ ProN W3\\",Roboto,\\"Droid Sans\\",\\"游ゴシック\\",YuGothic,\\"メイリオ\\",Meiryo,sans-serif;background:#db7093}</style><script>window.oc = window.oc || {};oc.cmd = oc.cmd || [];oc.cmd.push(function(oc){oc.events.fire(\\\\'oc:cssDidMount\\\\', \\\\'.awesome__p4GK2pPl{font-family:\\"Hiragino Sans\\",\\"ヒラギノ角ゴシック\\",\\"Hiragino Kaku Gothic ProN\\",\\"ヒラギノ角ゴ ProN W3\\",Roboto,\\"Droid Sans\\",\\"游ゴシック\\",YuGothic,\\"メイリオ\\",Meiryo,sans-serif;background:#db7093}\\\\');});</script>' }"`;
